added pt-BR language

jonasw8 opened the pull request.

jonasw8 committed:
Adicionada a tradução pt-BR

jonasw8 committed:
added pt-BR language

pyquick commented:
WOW!

perez987 commented:
@jonasw8 
Not enough to get the localization.

1. Get the attached English localization file, it's the latest, complete your Brazilian localization.

[Localizable.strings.zip](https://github.com/user-attachments/files/21598986/Localizable.strings.zip)

2.  All localization files have credits to contributors, add yourself in this way at least on Brazilian file:

```swift
"contributor1" = "laobamac - Desenvolvedor";
"contributor2" = "perez987 - Traduções parciais para espanhol e inglês";
"contributor3" = "Anto65 - Tradução parcial para o italiano";
"contributor4" = "lshbluesky - Tradução parcial para o coreano";
"contributor5" = "jonasw8 - Tradução parcial para o brasileiro";
```

3. LanguageManager.swift

```swift
    func availableLanguages() -> [String] {
        return ["en", "zh-Hans", "zh-Hant", "es", "it", "ko", "pt"] // add pt here
    }
    
    func displayName(for language: String) -> String {
        switch language {
        case "en": return "English"
        case "zh-Hans": return "简体中文"
        case "zh-Hant": return "繁体中文"
        case "es": return "Español"
        case "it": return "Italian"
        case "ko": return "Korean"
        case "pt": return "Brazilian" // add this line
        default: return language
        }
```

4. AboutView.swift

```swift
    let contributors = [
        "contributor1".localized,
        "contributor2".localized,
        "contributor3".localized,
        "contributor4".localized,
        "contributor5".localized, // add this line
    ]
```
Do the changes and update the PR.

Did it work the SimpleLoader for Ventura that I sent to you?

jonasw8 committed:
update about view

jonasw8 committed:
update language manager

laobamac commented:
Hey, bro, I just saw the email reminder from Github. I will check these PRs in a few days and add some functions at the same time. I will link GithubAPI to realize that everyone can share their useful Patchsets in next version. Recently, I have a little gastrointestinal cold and still have a fever. I can't check these codes for the time being.

Best regards!

perez987 commented:
@jonasw8 
In localization files:

```swift
"contributor1" = "laobamac - Developer";
"contributor2" = "perez987 - Partial Spanish and English translation";
"contributor3" = "Anto65 - Partial Italian translation";
"contributor4" = "lshbluesky - Partial Korean translation";
"contributor5" = "jonasw8 - Partial Brazilian translation";
```

jonasw8 committed:
updated

jonasw8 commented:
> @jonasw8 Not enough to get the localization.
> 
>     1. Get the attached English localization file, it's the latest, complete your Brazilian localization.
> 
> 
> [Localizable.strings.zip](https://github.com/user-attachments/files/21598986/Localizable.strings.zip)
> 
>     2. All localization files have credits to contributors, add yourself in this way at least on English and Brazilian files:
> 
> 
> ```swift
> "contributor1" = "laobamac - Developer";
> "contributor2" = "perez987 - Partial Spanish and English translation";
> "contributor3" = "Anto65 - Partial Italian translation";
> "contributor4" = "lshbluesky - Partial Korean translation";
> "contributor5" = "jonasw8 - Partial Brazilian translation";
> ```
> 
>     3. LanguageManager.swift
> 
> 
> ```swift
>     func availableLanguages() -> [String] {
>         return ["en", "zh-Hans", "zh-Hant", "es", "it", "ko", "pt"] // add pt here
>     }
>     
>     func displayName(for language: String) -> String {
>         switch language {
>         case "en": return "English"
>         case "zh-Hans": return "简体中文"
>         case "zh-Hant": return "繁体中文"
>         case "es": return "Español"
>         case "it": return "Italian"
>         case "ko": return "Korean"
>         case "pt": return "Brazilian" // add this line
>         default: return language
>         }
> ```
> 
>     4. AboutView.swift
> 
> 
> ```swift
>     let contributors = [
>         "contributor1".localized,
>         "contributor2".localized,
>         "contributor3".localized,
>         "contributor4".localized,
>         "contributor5".localized, // add this line
>     ]
> ```
> 
> Do the changes and update the PR.
> 
> Did it work the SimpleLoader for Ventura that I sent to you?

I made all the necessary changes that you mentioned.

The Ventura version didn't work, I updated my system to Sequoia 15.6

perez987 commented:
@jonasw8 

Please change this in your Brazilian file:

```swift
"contributor1" = "laobamac - 开发者";
"contributor2" = "perez987 - 提供英语翻译";
"contributor3" = "Anto65 - Partial Italian translation";
"contributor4" = "lshbluesky - Partial Korean translation";
"contributor5" = "JonasW8 - Brazilian Portuguese";
```

to this (fix the text if it isn't accurate):

```swift
"contributor1" = "laobamac - Desenvolvedor";
"contributor2" = "perez987 - Traduções parciais para espanhol e inglês";
"contributor3" = "Anto65 - Tradução parcial para o italiano";
"contributor4" = "lshbluesky - Tradução parcial para o coreano";
"contributor5" = "jonasw8 - Tradução parcial para o brasileiro";
```

There is 3 duplicate lines, remove one of each duplicate:

- lines 46 and 132: "author" = "Autor";
- lines 40 and 111: "cancel" = "Cancelar";
- lines 41 and 90: "about" = "Sobre";

jonasw8 commented:
> @jonasw8
> 
> Please change this in your Brazilian file:
> 
> ```swift
> "contributor1" = "laobamac - 开发者";
> "contributor2" = "perez987 - 提供英语翻译";
> "contributor3" = "Anto65 - Partial Italian translation";
> "contributor4" = "lshbluesky - Partial Korean translation";
> "contributor5" = "JonasW8 - Brazilian Portuguese";
> ```
> 
> to this (fix the text if it isn't accurate):
> 
> ```swift
> "contributor1" = "laobamac - Desenvolvedor";
> "contributor2" = "perez987 - Traduções parciais para espanhol e inglês";
> "contributor3" = "Anto65 - Tradução parcial para o italiano";
> "contributor4" = "lshbluesky - Tradução parcial para o coreano";
> "contributor5" = "jonasw8 - Tradução parcial para o brasileiro";
> ```
> 
> There is 3 duplicate lines, remove one of each duplicate:
> 
>     * lines 46 and 132: "author" = "Autor";
> 
>     * lines 40 and 111: "cancel" = "Cancelar";
> 
>     * lines 41 and 90: "about" = "Sobre";

The duplicate lines are in accordance with the Localizable.strings you sent me
lines 46 and 131 = author
lines 40 and 110 = cancel
lines 41 and 89 = about

perez987 commented:
@jonasw8 
It was a mistake on my part. I noticed and fixed it after sending you the file. They was in the first English localization file and so in our translations. Even the Korean file has them.

jonasw8 committed:
Update Localizable.strings

jonasw8 commented:
> @jonasw8 It was a mistake on my part. I noticed and fixed it after sending you the file. They was in the first English localization file and so in our translations. Even the Korean file has them.

updated
